lib/wpc_proto/internal_modules/proto_config.c

```diff
@@ -15,8 +15,194 @@
+{
+    app_res_e res = APP_RES_INTERNAL_ERROR;
+    if (Not_condition)
+    {
+        // Values are not defined
+        return true;
+    }
+    if (var2 == NULL)
+    {
+        res = ((func_1_param) f)(var1);
+    }
+    else
+    {
+        res = ((func_2_param) f)(var1, var2);
+    }
+
+    if (res != APP_RES_OK)
+    {
+        LOGE("Cannot get %s from node\n", var_name);
+        return false;
+    }
+
+    return true;
+}
+
+static bool initialize_unmodifiable_variables()
+{
+    bool res = true;
+
+    res &= get_value_from_node(WPC_get_stack_profile, &m_sink_config.stack_profile, NULL, "Stack profile", false);
+    res &= get_value_from_node(WPC_get_hw_magic, &m_sink_config.hw_magic, NULL, "Hw magic", false);
+    res &= get_value_from_node(WPC_get_mtu, &m_sink_config.max_mtu, NULL, "MTU", false);
+    res &= get_value_from_node(WPC_get_pdu_buffer_size, &m_sink_config.pdu_buffer_size, NULL, "PDU Buffer Size", false);
+    res &= get_value_from_node(WPC_get_channel_limits, &m_sink_config.ch_range_min, &m_sink_config.ch_range_max,
+                               "Channel Range", false);
+    res &= get_value_from_node(WPC_get_access_cycle_limits, &m_sink_config.ac_limit_min, &m_sink_config.ac_limit_max,
+                               "AC Range", false);
+    res &= get_value_from_node(WPC_get_app_config_data_size, &m_sink_config.app_config_max_size, NULL,
+                               "App Config Max size", false);
+
+    if (WPC_get_firmware_version(m_sink_config.version) == APP_RES_OK)
+    {
+        LOGI("Stack version is: %d.%d.%d.%d\n",
+             m_sink_config.version[0],
+             m_sink_config.version[1],
+             m_sink_config.version[2],
+             m_sink_config.version[3]);
+    }
+    else
+    {
+        res = false;
+    }
+
+    if (!res)
+    {
+        LOGE("All the static settings cannot be read\n");
+    }
+
+    return res;
+}
+
+static bool initialize_variables()
+{
+    _Static_assert(member_size(wp_NodeRole, flags) == (2 * sizeof(wp_NodeRole_RoleFlags)) );
+
+    bool res = true;
+    app_role_t role = 0;
+    pb_size_t flags_count = 0;
+
+    res &= get_value_from_node(WPC_is_cipher_key_set, &m_sink_config.CipherKeySet, NULL,
+                               "Cipher key set", false);
+    res &= get_value_from_node(WPC_is_authentication_key_set, &m_sink_config.AuthenticationKeySet, NULL,
+                               "Authentication key set", false);
+    res &= get_value_from_node(WPC_get_stack_status, &m_sink_config.StackStatus, NULL, "Stack Status", false);
+
+    res &= get_value_from_node(WPC_get_access_cycle_range, &m_sink_config.ac_range_min_cur, &m_sink_config.ac_range_max_cur,
+                               "Current access cycle range", false);
+
+    res &= get_value_from_node(WPC_get_node_address, &m_sink_config.node_address, NULL,
+                               "Network address", m_sink_config.StackStatus & APP_STACK_NODE_ADDRESS_NOT_SET);
```

Comment:
Why using m_sink_config.StackSatus here?

Reply:
I remove all the usage of stackStatus bits (except stack on/off).